### PR TITLE
perf: migrate to @oclif/core

### PR DESCRIPTION
**Pull request by john-u**
- [x] decide on [which oclif flags to keep](https://github.com/oclif/core/blob/main/MIGRATION.md#restore--h--v-and-version)

### CODE REVIEW

File: packages/cli/README.md [hunk 1]
```diff
@@ -233,1620 +232,2101 @@ get a specific app or a list of apps
 
 ```
 USAGE
-  $ smartthings apps [ID]
+  $ smartthings apps [ID] [-h] [-p <value>] [-t <value>] [--language <value>] [--indent <value>] [-j] [-y]
+    [-o <value>] [--compact] [--expanded] [--type <value>] [--classification <value>] [-v]
```
Comment (john-u) [inline packages/cli/README.md:236]: The changes to readme output are from migrating away from `@oclif/dev-cli` to `oclif` instead.

File: packages/cli/README.md [hunk 2]
```diff
@@ -188,7 +188,6 @@ that map to the API spec.
 * [`smartthings edge:drivers:uninstall [DRIVERID]`](#smartthings-edgedriversuninstall-driverid)
 * [`smartthings generate:java`](#smartthings-generatejava)
 * [`smartthings generate:node`](#smartthings-generatenode)
-* [`smartthings help [COMMAND]`](#smartthings-help-command)
```
[the remark this answers is not in the document]
Comment (rossiam) [reply packages/cli/README.md:191]: I don't think it's necessary.

File: packages/cli/src/hooks/init/init-config.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Hook } from '@oclif/config'
+import type { Hook } from '@oclif/core'
```
Comment (rossiam) [inline packages/cli/src/hooks/init/init-config.ts:1]: Was this intentional?

**Review by rossiam — APPROVED** [review]

File: packages/cli/src/hooks/init/init-config.ts (hunk 3 above)
Comment (john-u) [reply packages/cli/src/hooks/init/init-config.ts:1]: Yes, I thought I'd change this to a [type-only import](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export).

**Pull request merged**